src/structures/gdt.rs

```diff
@@ -61,57 +93,70 @@ impl GlobalDescriptorTable {
 }
 
 impl<const MAX: usize> GlobalDescriptorTable<MAX> {
-    /// Creates an empty GDT which can hold `MAX` number of [`Descriptor`]s.
+    /// Creates an empty GDT which can hold `MAX` number of [`Entry`]s.
     #[inline]
     pub const fn empty() -> Self {
         // TODO: Replace with compiler error when feature(generic_const_exprs) is stable.
         assert!(MAX > 0, "A GDT cannot have 0 entries");
         assert!(MAX <= (1 << 13), "A GDT can only have at most 2^13 entries");
+        const NULL: Entry = Entry::new(0);
         Self {
-            table: [0; MAX],
+            table: [NULL; MAX],
             len: 1,
         }
     }
 
     /// Forms a GDT from a slice of `u64`.
     ///
-    /// # Safety
+    /// This method allows for creation of a GDT with malformed or invalid
+    /// entries. However, it is safe because loading a GDT with invalid
+    /// entires doesn't do anything until those entries are used. For example,
+    /// [`CS::set_reg`] and [`load_tss`](crate::instructions::tables::load_tss)
+    /// are both unsafe for this reason.
     ///
-    /// * The user must make sure that the entries are well formed
-    /// * Panics if the provided slice has more than `MAX` entries
+    /// Panics if:
+    /// * the provided slice has more than `MAX` entries
+    /// * the provided slice is empty
+    /// * the first entry is not zero
+    #[cfg_attr(not(feature = "instructions"), allow(rustdoc::broken_intra_doc_links))]
     #[inline]
-    pub const unsafe fn from_raw_slice(slice: &[u64]) -> Self {
+    pub const fn from_raw_entries(slice: &[u64]) -> Self {
```

Comment:
What's the reason for not using a `&[Entry]` slice here? We would need to make `Entry::new` public for that, but creating an `Entry` from an `u64` is also possible using this function.

Reply:
This was mostly to preserve the previous setup, where it seemed like people wanted a way to convert an array of raw values (that they potentially get from somewhere else) to a GDT.

If we accepted `Entry`s and a user had a `&[u64]`, even if `Entry::new` was public, they would either need to copy the slice to create a `&[Entry]`. With #381 (and entry having interior mutability), this also lets us initialize the GDT from truly read-only data. There's not a reason for the data we are initializing the GDT with to have interior mutability.

Reply:
Also on a more practical level, if we make `Entry` an `AtomicU64` we wouldn't be able to have this method be `const fn` anymore, as that would require reading from the atomic in a `const fn` which isn't currently supported.

Reply:
Makes sense, thanks!